server/events/project_command_builder.go

```diff
@@ -345,6 +351,27 @@ func (p *DefaultProjectCommandBuilder) buildProjectPlanCommand(ctx *command.Cont
 		workspace = cmd.Workspace
 	}
 
+	if p.StrictPlanFileList {
```

Comment:
Will this not only check for `-d` (directory) outside of the current PR's contents but also `-p` (repo atlantis.yaml defined project) as well ?

Reply:
Hmm, yes it does seem like we need to not only check if `cmd.RepoRelDir != ""` but also `cmd.ProjectName != ""`.

Please don't use this exact code but it seems like this is the way to get the repo relative directory from the project. ([source](https://github.com/runatlantis/atlantis/blob/b753924befd9be547e3ca2dfe82fedeaa623fcc1/server/events/project_command_builder.go#L606-L621))

```go
import (
	"github.com/runatlantis/atlantis/server/core/config/valid"
)

// ...

	if cmd.ProjectName != "" {
		var projCfg valid.MergedProjectCfg
		repoRelDir = projCfg.RepoRelDir
	}
```

Reply:
@Fabianoshz I left this comment unresolved so we can account for this.

Reply:
I agree we need a -p flag for this too, since many project can have many dirs and you could potentially bypass this restrictions just by using -p project name that matches any other dir

Reply:
Added the logic for the -p flag under the `--strict-plan-file-list`, if you guys believe this should be under a different flag or the implementation is not good enough, please let me know.

Reply:
Hmm the plan word seems counter intuitive because this will also restrict applies? Unsure on a better name.

cc @runatlantis/maintainers

Reply:
This will not interact with applies, at least from my understanding the `buildProjectPlanCommand()` will only answer plan commands, my expectation is that by doing this users will not be able to plan anything outside their defined scope, which will lead to a situation where the apply will never be available for files outside their scope too because the planfile will never be created.

Reply:
right. I agree with that.

Since it will affect both, even though you can only apply what you plan, I'm suggesting to strip the `plan` portion of the name of the flag so it's simply `--strict-file-list` or `--restrict-file-list`.

cc: @runatlantis/maintainers thoughts on the name of this flag?

Reply:
--restrict-file-list sound better I think, the word Plan makes it a bit confusing since it will apply to both actions

Reply:
but I can live with both

Reply:
Changed to `--restrict-file-list`